openc3/spec/accessors/binary_accessor_spec.rb

```diff
@@ -23,9 +23,137 @@
+        expect(item3.bit_offset).to eq 256
+
+        @packet.write("item2", "Small")
+        expect(@packet.buffer).to eql "\x00\x00\x00\x0FThis is Awesome\x00\x05Small\x05\x01\x02\x03\x04\x05"
+        expect(item1_length.bit_offset).to eq 0
+        expect(item1.bit_offset).to eq 32
+        expect(item2_length.bit_offset).to eq 152
+        expect(item2.bit_offset).to eq 168
+        expect(item3_length.bit_offset).to eq 208
+        expect(item3.bit_offset).to eq 216
+
+        @packet.write("item1", "Yo")
+        expect(@packet.buffer).to eql "\x00\x00\x00\x02Yo\x00\x05Small\x05\x01\x02\x03\x04\x05"
+        expect(item1_length.bit_offset).to eq 0
+        expect(item1.bit_offset).to eq 32
+        expect(item2_length.bit_offset).to eq 48
+        expect(item2.bit_offset).to eq 64
+        expect(item3_length.bit_offset).to eq 104
+        expect(item3.bit_offset).to eq 112
+
+        # Recalculates offsets when buffer is written
+        @packet.buffer = "\x00\x00\x00\x03Ok!\x00\x04Cool\x02\xA5\x5A"
+        expect(item1_length.bit_offset).to eql 0
+        expect(item1.bit_offset).to eql 32
+        expect(item2_length.bit_offset).to eql 56
+        expect(item2.bit_offset).to eql 72
+        expect(item3_length.bit_offset).to eql 104
+        expect(item3.bit_offset).to eql 112
+        expect(@packet.read("item1_length")).to eql 3
+        expect(@packet.read("item1")).to eql "Ok!"
+        expect(@packet.read("item2_length")).to eql 4
+        expect(@packet.read("item2")).to eql "Cool"
+        expect(@packet.read("item3_length")).to eql 2
+        expect(@packet.read("item3")).to eql [0xA5, 0x5A]
+      end
+
+      it "can define multiple variable sized items with non-zero value offsets" do
+        item1_length = @packet.append_item("item1_length", 32, :INT)
+        expect(item1_length.bit_offset).to eql 0
+        expect(@packet.buffer).to eql "\x00\x00\x00\x00"
+        item1 = @packet.append_item("item1", 0, :STRING)
+        @packet.write("item1_length", 4) # Lengths must be initialized to zero equivalent
+        expect(@packet.buffer).to eql "\x00\x00\x00\x04"
+        item1.variable_bit_size = {'length_item_name' => 'item1_length', 'length_value_bit_offset' => -32, 'length_bits_per_count' => 8}
+        expect(item1.bit_offset).to eql 32
+        item2_length = @packet.append_item("item2_length", 16, :UINT)
+        expect(item2_length.bit_offset).to eql 32
+        expect(@packet.buffer).to eql "\x00\x00\x00\x04\x00\x00"
+        item2 = @packet.append_item("item2", 0, :STRING)
+        expect(@packet.buffer).to eql "\x00\x00\x00\x04\x00\x00"
+        @packet.write("item2_length", 8) # Lengths must be initialized to zero equivalent
+        item2.variable_bit_size = {'length_item_name' => 'item2_length', 'length_value_bit_offset' => -64, 'length_bits_per_count' => 8}
+        expect(item2.bit_offset).to eql 48
+        expect(@packet.buffer).to eql "\x00\x00\x00\x04\x00\x08"
```

Comment:
I'm not sure what's going on with this test. How did we get the 4 and 8? Negative offset is a positive value? What happens if you have a value offset of +32?

Reply:
This test is handling the case where the value field includes itself in the count (so that has to be subtracted out)
A positive offset would be something like CCSDS where a value of 0 = 1 byte = 8 bits.  Don't have a test for that yet.